### DISCUSSION TITLE
Reverse `config-cycle` binding

### DISCUSSION THREAD

**Discussion by mrossinek** [qutebrowser/qutebrowser#7667] [Q&A] [answered]
I have a key binding like the following in my config:
```python
config.bind("<Ctrl+=>", "config-cycle zoom.default 100% 125% 150% 175% 200%")
```
This allows me to cycle through multiple zoom settings e.g. when switching from a low-res internal to high-res external screen. So far, when switching back, I always need to hit the binding multiple times to "cycle" through when e.g. wanting to go from 150% to 100%. So I was wondering: is there a way to make the reverse mapping of this and have them aligned? E.g. add a binding for `Ctrl + -` which knows the current zoom state and respects the existence of both bindings at the same time?

**Comment by The-Compiler** [comment]
Sorry for the late answer, this must have slipped through the cracks!

In the general case, this is not possible - if you see other use cases for this, I'd suggest you open an issue for a feature request, or even a PR. I don't think it should be difficult to add an option which uses `-1` instead of `+ 1` here:

https://github.com/qutebrowser/qutebrowser/blob/346a39ba14bdc05ce21dc1233027803dcaf34cc6/qutebrowser/config/configcommands.py#L236

In this specific case, however, couldn't you just set `zoom.levels` and use `zoom-in` / `zoom-out` instead?

**Reply by mrossinek** [reply]
Thanks for the reply! The idea here is related to https://github.com/qutebrowser/qutebrowser/discussions/7396 where I would like to set the zoom level for all open tabs at the same time. The use case being e.g. when switching from a high-res to low-res monitor.
This seems fairly easy to configure in this case, I might open a PR when I have some more time on my hands :+1:

**Reply by The-Compiler** [reply]
I opened an issue: #7683

**Comment by mrossinek** [comment, marked as answer]
Actually, this is already possible because qutebrowser can correctly handle the config state. Not sure what I did wrong in my previous tests on this... I already wrote this [here](https://github.com/qutebrowser/qutebrowser/issues/7683#issuecomment-1528816779) but am repeating it here for the sake of completeness:
```
config.bind("<Ctrl+=>", "config-cycle zoom.default 100% 125% 150% 175% 200%")
config.bind("<Ctrl+->", "config-cycle zoom.default 200% 175% 150% 125% 100%")
```
